code/modules/mob/living/carbon/xenomorph/castes/wraith/abilities_wraith.dm

```diff
@@ -0,0 +1,494 @@
+	var/duration = WRAITH_BANISH_BASE_DURATION //Set the duration
+
+	if(isxeno(banishment_target) ) //We halve the max duration for living non-allies
+		var/mob/living/carbon/xenomorph/X = banishment_target
+		if(!X.issamexenohive(ghost)) //Enemy hive
+			duration *= 0.5
+
+	else if(isliving(banishment_target) ) //We halve the max duration for living non-allies
+		duration *= 0.5
+
+	banishment_target.visible_message("<span class='warning'>Space abruptly twists and warps around [banishment_target.name] as it suddenly vanishes!</span>", \
+	"<span class='highdanger'>The world around you reels, reality seeming to twist and tear until you find yourself trapped in a forsaken void beyond space and time.</span>")
+	playsound(banished_turf, 'sound/weapons/emitter2.ogg', 50, 1) //this isn't quiet
+
+	to_chat(ghost,"<span class='xenodanger'>We have banished [banishment_target.name] to nullspace for [duration * 0.1] seconds.</span>")
+
+	addtimer(CALLBACK(src, .proc/banish_warning), duration * 0.7) //Warn when Banish is about to end
+	addtimer(CALLBACK(src, .proc/banish_deactivate), duration)
+
+	succeed_activate()
+	add_cooldown()
+
+	GLOB.round_statistics.wraith_banishes++
+	SSblackbox.record_feedback("tally", "round_statistics", 1, "wraith_banishes") //Statistics
+
+/datum/action/xeno_action/activable/banish/proc/banish_warning()
+
+	if(!banished_turf || !banishment_target)
+		return
+
+	to_chat(owner,"<span class='highdanger'>Our banishment target [banishment_target.name] is about to return!</span>")
+	owner.playsound_local(owner, 'sound/voice/hiss4.ogg', 50, 0, 1)
+
+
+/datum/action/xeno_action/activable/banish/proc/banish_deactivate()
+	SIGNAL_HANDLER
+
+	if(!banished_turf || !banishment_target)
+		return
+
+	banishment_target.forceMove(banished_turf)
+	banishment_target.resistance_flags = initial(banishment_target.resistance_flags)
+	teleport_debuff_aoe(banishment_target) //Debuff/distortion when we reappear
+	banishment_target.add_filter("wraith_banishment_filter", 3, list("type" = "blur", 5)) //Cool filter appear
+	addtimer(CALLBACK(banishment_target, /atom.proc/remove_filter, "wraith_banishment_filter"), 1 SECONDS) //1 sec blur duration
+
+	var/mob/living/living_target
+	if(isliving(banishment_target))
+		living_target = banishment_target
+		living_target.status_flags = initial(living_target.status_flags) //Remove stasis and temp invulerability
+		living_target.SetStasis(remove = TRUE)
+		living_target.Stun(1 SECONDS) //Short stun upon snap back to reality
+
+	banishment_target.visible_message("<span class='warning'>[banishment_target.name] abruptly reappears!</span>", \
+	"<span class='warning'>You suddenly reappear back in what you believe to be reality. It takes you a moment to regain your bearings.</span>")
+
+	to_chat(owner, "<span class='xenodanger'>Our target [banishment_target.name] has returned to reality at [get_area(banishment_target)] (X: [banishment_target.x], Y: [banishment_target.y])</span>") //Always alert the Wraith
+
+	if(portal)
```

Comment:
Excess if

Reply:
This deletes the Banish singularity if the Wraith recalls its Banish target from Brazil early.